lib/confluence/aggressive_removal.es6.js

```diff
@@ -0,0 +1,144 @@
+'use strict';
+
+require('./aggressive_removal_data.es6.js');
+
+foam.CLASS({
+  package: 'org.chromium.apis.web',
+  name: 'AggressiveRemoval',
+  extends: 'org.chromium.apis.web.MetricComputer',
+  documentation: `AggressiveRemoval is a class that computes the aggressive
+    removal metrics from browserApiDAO. It shows the number of APIs
+    that is removed in this browser at least 2 years ago but still exits in
+    other browsers.`,
+  requires: [
+    'org.chromium.apis.web.Browser',
+    'org.chromium.apis.web.WebInterface',
+    'org.chromium.apis.web.BrowserWebInterfaceJunction',
+    'org.chromium.apis.web.AggressiveRemovalData',
+    'foam.mlang.ExpressionsSingleton',
+    'foam.dao.ArraySink',
+    'foam.dao.EasyDAO',
+  ],
+  properties: [
+    {
+      name: 'aggressiveRemovalDAO',
+      documentation: `This is a DAO that contains AggressiveRemovalData.`,
+      final: true,
+      factory: function() {
+        return this.EasyDAO.create({
+          name: 'aggressiveRemovalDAO',
+          of: this.AggressiveRemovalData,
+          daoType: 'MDAO',
+        });
+      },
+    },
+  ],
+  methods: [
+    {
+      name: 'compute',
+      documentation: `Compute aggressive removal value for each browser in
+           browsers at the given date.`,
+      args: [
+        {
+          name: 'browsers',
+          typeName: 'Array of Browser',
+          documentation: `An array of brwsers. These browser are either
+              released at given date, or it is the most recent release
+              before given date.`,
+        },
+        {
+          name: 'date',
+          typeName: 'Date',
+          documentation: `The aggressive removal value will be calculated at
+              given date.`,
+        },
+      ],
+      code: function(browsers, date) {
+        for (let i = 0; i < browsers.length; i++) {
+          let browser = browsers[i];
+          let year = date.getFullYear();
+          let month = date.getMonth();
+          let day = date.getDate();
+          let oneYearAgo = new Date(year - 1, month, day);
+          let browserKeys = browsers.map((browser) => browser.browserKey);
+          let selectBrowsers = [
+            this.browserDAO.where(this.mlang.AND(
+              this.mlang.IN(this.Browser.BROWSER_KEY, browserKeys),
+              this.mlang.NEQ(this.Browser.BROWSER_NAME, browser.browserName)))
+            .select(),
+            this.browserDAO.where(this.mlang.AND(
+              this.mlang.EQ(this.Browser.BROWSER_NAME, browser.browserName),
+              this.mlang.EQ(this.Browser.OS_NAME, browser.osName),
+              this.mlang.LT(this.Browser.RELEASE_DATE, oneYearAgo)))
+            .select()];
+          Promise.all(selectBrowsers).then((result) => {
+            let currBrowsers = result[0].a;
+            let prevReleaseBrowsers = result[1].a;
+            // Return if there is insufficient of historical data.
+            if (prevReleaseBrowsers.length < 2) return;
```

Comment:
There is no way to get aggressive-removal data if we only have less than two version's object graph.